### DISCUSSION TITLE
beginner questions

### DISCUSSION THREAD

**Discussion by mvolkmann** [dataFude-app/dataFude#13] [Beta feedback]
Where can I find documentation that explains the kinds of values that can be passed to `os_log` that trigger dataTile updates?
I thought this would work, but it doesn't for me: `os_log("someVariable = \(someVariable)").
These do work though:

```swift
os_log("someVariable: \(someVariable)")
os_log("someVariable ' \(someVariable)'")
```

Is calling `os_log` the only thing that triggers dataTile updates?

Is the Mini version limited to just showing the total number of updates in one tile and showing one or two values in additional tiles?
Can we not show the total number of updates and show three values in tiles?

**Comment by icanzilb** [comment]
Hey @mvolkmann those are great questions. 

 - regarding the recognized patterns — open the app settings window and you will see a tab called "Automatic Tiles" which lists all the patterns that are recognized and let you turn on and off some of them. 
 - yes, the unified log is the way to drive dataTile's widgets. dataTile observes your log directly in the Simulator. You can use the `os_log()` function or any other of Apple's APIs to write to the log (https://developer.apple.com/documentation/os/logging). Do you have any other way in mind that would work better for your development workflow?
 - The Mini is currently limited to 3 tiles in total

**Reply by mvolkmann** [reply]
Thanks!

On "recognized patterns", do you know why this isn't working? `os_log("tripled = \(count * 3)")`
If I change the `=` to `:` then it works.

I'm fine with using `os_log` to trigger tile updates. I was just curious if there were other ways I should be aware of.

Is there a way to tell Mini not to use the first tile for displaying the total number of updates so I can use all three tiles to display values?

**Reply by icanzilb** [reply]
The Activity is always the first tile regardless of Mini or Pro. If you are on the Mini the first pattern using colons is pre-checked.
